### ISSUE TITLE
CORS Policy in REST API

### ISSUE THREAD

**Issue by mgordel** [golemfactory/yagna#2000]
Unable to connect to Rest API from browser due to CORS policy.

In most browsers, the `Access-Control-Allow-Origin` header must be set to send a query to the yagna Rest API and receive a correct response. Now all responses without these header are blocked by browser.

Perhaps adding the `Access-Control-Allow-Origin: *` header in response would solve the problem, or maybe the user should be able to configure this option for a specific origin?

- Talk with @mgordel about requirements and how should it work
- Header should be configurable from ENV (list of domains names)
- Headers list should be configurable per identity

(Maybe should be implemented as middlware)

**Comment by pwalski** [comment]
IMO there will be a need for middleware solution.
Browser support means hosting some statics which will not be done by yagna, so there will be a necessity to use some API gateway to route between API requests and requests for statics.
Additionally [WebSocket related task](https://github.com/golemfactory/ya-service-bus/issues/46) will result in a support of WebSocket in GSB. The problem is GSB does not share port with Yagna API so it will be an additional necessity to have an API gateway (especially because GSB can work standalone, and in case of _browser support deployment_ it may make sense to have it deployed as a service separated from `yagna`).

**Comment by pwalski** [comment]
This task will have a low priority and most likely will be closed.

Adding CORS to Yagna only makes easier development of requestor webapps which do not use GSB API.
Yagna GSB API will be accessible from a different address (port) than Yagna REST API, so in order to make them both accessible to webapp Yagna will need to sit in front of API gateway (locally it could be e.g. nginx) which can be configured to add CORS policy.

Problem of local setup will be easy to solve using e.g. `docker-compose.yml` starting yagna + nginx.

In a real world deployment yagna will always sit in front of a middle layer handling https and routing between requests to webapp statics, Yagna API, and Yagna GSB API. Additionally in future API Gateway most likely will need to handle authorization and load balancing. API Gateway is an usual place for a CORS policy config in such case.

Furthermore adding CORS to Yagna would be a source of config errors in future because Yagna REST API will not handle https, so the `YAGNA_API_URL` and CORS URL would not only have a different address but also a different protocol.

**Comment by pwalski** [comment]
Alternatively this task could result in a distribution of `yagna` as a Docker image starting it as a requestor together with server/reverse proxy in front of it.
Image would export port 80 and provide mapping for "api.localhost" pointing to Yagna API and "gsb.localhost" pointing to GSB.

**Comment by pwalski** [comment]
Duplicate of #2360
